dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/HelperInjector.java

```diff
@@ -135,12 +147,23 @@ private Map<String, Class<?>> injectClassLoader(
       final ClassLoader classLoader, final Map<String, byte[]> classnameToBytes) {
     INJECTING_HELPERS.begin();
     try {
-      return new ClassInjector.UsingReflection(classLoader).injectRaw(classnameToBytes);
+      ProtectionDomain protectionDomain = createProtectionDomain(classLoader);
+      return new ClassInjector.UsingReflection(classLoader, protectionDomain)
+          .injectRaw(classnameToBytes);
     } finally {
       INJECTING_HELPERS.end();
     }
   }
 
+  private ProtectionDomain createProtectionDomain(final ClassLoader classLoader) {
+    if (useAgentCodeSource) {
+      CodeSource codeSource = HelperInjector.class.getProtectionDomain().getCodeSource();
+      return new ProtectionDomain(codeSource, null, classLoader, null);
+    } else {
+      return null;
```

Comment:
```suggestion
      return ClassLoadingStrategy.NO_PROTECTION_DOMAIN;
```
This constant boils down to the same literal, i.e. `null` but is more descriptive

Reply:
Good point, replaced literal with the constant